Sprint 12

Neenjahw opened the pull request:
Спринт 12 на ревью

Neenjahw committed:
sprint_08
1. Сверстан экран LaunchScreen
2. Сверстан экран главного приложения

Neenjahw committed:
no message

Neenjahw committed:
Добавлен градиент для dateLable

Neenjahw committed:
Спринт 9 на ревью

Neenjahw committed:
no message

Neenjahw committed:
Добавил констрейнты ширины и высоты для кнопки поделиться

Neenjahw committed:
Сдаем на ревью спринт 10

Neenjahw committed:
Изменил путь до info.plist

Neenjahw committed:
Добавлен super.viewDidLoad() в WebViewController
Сделан class ImageListCell final

Neenjahw committed:
На ревью 11 спринт

Neenjahw committed:
1. Изменил fatalError на assertionFailure в AuthViewController
2. Сделал свойство scrollView приватным в SingleImageViewController
3. Избавился от force unwrap в ImageListViewController
4. Перенес свойство static let unsplashAuthorizeURLString в файл Constants

Neenjahw committed:
Реализовал постраничную загрузку данных для ленты фотографий

Neenjahw committed:
Спринт 12 на ревью

DanilaG commented on ImageFeed/ImagesList/Services/ImageListService.swift:
```diff
@@ -0,0 +1,143 @@
+        let created_at: Date?
+        let size: CGSize
+        let welcomeDescription: String?
+        let thumbImageURL: String
+        let largeImageURL: String
```
Давай хранить Url используя тип данных `URL`

DanilaG commented on ImageFeed/ImagesList/Services/ImageListService.swift:
```diff
@@ -0,0 +1,143 @@
+        let largeImageURL: String
+        var isLiked: Bool
+        
+        init(from photoResult: PhotoResult) {
```
Обычно, инит для структуры делают на основе тех полей, с которыми она работает, а мепинг одной структуры в другую выносят в наружу. Так как это повышает тестируемость кода. Так, сейчас, при написании теста где нужна структура Photo придётся для её инита создовать PhotoResult

DanilaG commented on ImageFeed/ImagesList/ImagesListView/GradientView.swift:
```diff
@@ -0,0 +1,46 @@
+    
+    //MARK: - Private constants
+    private enum UIConstants {
+        static let startGradientColor = UIColor(red: 26/255, green: 27/255, blue: 34/255, alpha: 0.2).cgColor
+        static let middleGradientColor = UIColor(red: 26/255, green: 27/255, blue: 34/255, alpha: 0.2).cgColor
+        static let endGradientColor = UIColor(red: 26/255, green: 27/255, blue: 34/255, alpha: 0.01).cgColor
```
Давай сохраним цвета в assets и будем брать их от туда. Тогда по коду будет зразу понятно, что это за цвета, а изменять их нужно будет в одном месте для всего проекта)

Neenjahw committed:
Исправил замечания.
1. Сделал DateFormatter синглтоном (аналогично и для configCell)
2. Сделал в методе fetchPhotosNextPage проверку, если запрос уже выполняется, завершать выполнение функции
3. При выходе из приложения добавил Alert
4. Поменял тип данных let thumbImageURL: String на let thumbImageURL: URL
5. Использовал вместо инициализатора для Photo, преобразование через метод toPhoto
6. Избавился от UIConstants в GradientView, заменив на цвет, который находиться в Assets

DanilaG commented on ImageFeed/ImagesList/Services/ImageListService.swift:
```diff
@@ -0,0 +1,118 @@
+    static let didChangeNotification = Notification.Name(rawValue: "ImagesListServiceDidChange")
+    
+    static let isoDateFormatter: ISO8601DateFormatter = {
+           let dateFormatter = ISO8601DateFormatter()
+           return dateFormatter
+       }()
```
Давай поправим отступы

Neenjahw committed:
no message
